File: src/03.FrontEnd/04.FrontEnd.Logics/Modules/MasterData/Countries/GetCountries/GetCountriesQuery.cs
using SolutionTemplate2.Shared.Dto.Modules.MasterData.Countries.GetCountries;

namespace SolutionTemplate2.FrontEnd.Logics.Modules.MasterData.Countries.GetCountries;

public sealed record GetCountriesQuery : IRequest<GetCountriesResponse>
{
}

public sealed class GetCountriesQueryHandler(IBackEndApiService backEndApiService)
    : IRequestHandler<GetCountriesQuery, GetCountriesResponse>
{
    public async Task<GetCountriesResponse> Handle(GetCountriesQuery request, CancellationToken cancellationToken)
    {
        var restRequest = new RestRequest(GetCountriesRoute.ResourceUri, Method.Get);

        return await backEndApiService.SendRequestAsync<GetCountriesResponse>(restRequest, cancellationToken);
    }
}

File: src/04.FrontEndReact/04.FrontEndReact.Logics/Modules/MasterData/Countries/GetCountries/GetCountriesQuery.ts
import type { GetCountriesResponse } from "@services/Dto/Modules/MasterData/Countries/GetCountries/GetCountriesResponse";
import { GetCountriesRoute } from "@services/Dto/Modules/MasterData/Countries/GetCountries/GetCountriesRoute";
import { getBackEndApiService } from "@infrastructure/BackEndApi/ConfigureBackEndApi";

export async function sendGetCountriesQuery(): Promise<GetCountriesResponse> {
  return getBackEndApiService().sendRequestAsync<GetCountriesResponse>({
    resourceUri: GetCountriesRoute.ResourceUri,
    method: "GET",
  });
}
